server/filestore.go

```diff
@@ -6241,6 +6241,10 @@ func (o *consumerFileStore) Update(state *ConsumerState) error {
 		return fmt.Errorf("bad ack floor for stream")
 	}
 
+	// Replace our state.
+	o.mu.Lock()
```

Comment:
I mentioned that you should have move that at top of the function, that will also protect access to state.AckFloor, etc...